src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/DeleteUserProfileServiceImpl.java

```diff
@@ -30,16 +37,66 @@
     @Autowired
     private AuditService auditService;
 
+    @Autowired
+    private IdamFeignClient idamClient;
+
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse deleteByUserId(String userId) {
+        Response idamResponse = idamClient.deleteUser(userId);
+
+        if (idamResponse.status() == NO_CONTENT.value() || idamResponse.status() == NOT_FOUND.value()) {
+            UserProfile userProfile = validateUserStatus(userId);
+            return deleteUserProfiles(singletonList(userProfile));
+
+        } else {
+            UserProfilesDeletionResponse deletionResponse = new UserProfilesDeletionResponse();
+            deletionResponse.setMessage("IDAM Delete request failed for userId: " + userId);
+            deletionResponse.setStatusCode(idamResponse.status());
+            return deletionResponse;
+        }
+    }
+
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse deleteByEmailPattern(String emailPattern) {
+
+        List<UserProfile> userProfiles = userProfileRepository.findByEmailIgnoreCaseContaining(emailPattern);
+
+        if (userProfiles.isEmpty()) {
+            throw new ResourceNotFoundException("No User Profiles found for email pattern: " + emailPattern);
+        }
+
+        for (int i = userProfiles.size() - 1; i >= 0; i--) {
+            Response idamResponse = idamClient.deleteUser(userProfiles.get(i).getIdamId());
+
+            if (idamResponse.status() == NO_CONTENT.value() || idamResponse.status() == NOT_FOUND.value()) {
```

Comment:
NO_CONTENT is success scenario for delete from Idam ?

Reply:
Yes `NO_CONTENT` with `204` status is what is returned for successful deletion in IDAM.

_Please see their Swagger documentation [here](https://idam-api.aat.platform.hmcts.net/swagger-ui.html#/User_Role_Management)_